### DISCUSSION TITLE
How to build? - Extractor dev and NewPipe dev not compatible?

### DISCUSSION THREAD

**Discussion by 0verEngineer** [TeamNewPipe/NewPipe#10433] [Questions] [answered]
NewPipe uses properties that do not exist in the NewPipeExtractor, i have the newest dev Branch of both Projects.

I checked the git history of StreamInfoItem for the thumbnailUrl and the last change was in July 2022, what is going on here?

```
e: file:///home/julian/Projects/github/NewPipe/app/src/main/java/org/schabi/newpipe/database/playlist/PlaylistStreamEntry.kt:31:14 Unresolved reference: thumbnailUrl
e: file:///home/julian/Projects/github/NewPipe/app/src/main/java/org/schabi/newpipe/database/stream/StreamStatisticsEntry.kt:33:14 Unresolved reference: thumbnailUrl
e: file:///home/julian/Projects/github/NewPipe/app/src/main/java/org/schabi/newpipe/database/stream/model/StreamEntity.kt:70:61 Unresolved reference: thumbnailUrl
e: file:///home/julian/Projects/github/NewPipe/app/src/main/java/org/schabi/newpipe/database/stream/model/StreamEntity.kt:79:61 Unresolved reference: thumbnailUrl
e: file:///home/julian/Projects/github/NewPipe/app/src/main/java/org/schabi/newpipe/database/stream/model/StreamEntity.kt:96:14 Unresolved reference: thumbnailUrl
e: file:///home/julian/Projects/github/NewPipe/app/src/main/java/org/schabi/newpipe/local/subscription/SubscriptionFragment.kt:340:34 Unresolved reference: thumbnailUrl
e: file:///home/julian/Projects/github/NewPipe/app/src/main/java/org/schabi/newpipe/local/subscription/SubscriptionManager.kt:74:48 Unresolved reference: avatarUrl
e: file:///home/julian/Projects/github/NewPipe/app/src/main/java/org/schabi/newpipe/local/subscription/SubscriptionManager.kt:102:22 Unresolved reference: avatarUrl
e: file:///home/julian/Projects/github/NewPipe/app/src/main/java/org/schabi/newpipe/local/subscription/item/ChannelItem.kt:42:43 Unresolved reference: thumbnailUrl
```

**Comment by AudricV** [comment]
> Extractor dev and NewPipe dev not compatible?

Yes, they are not currently compatible, due to extractor breaking changes (TeamNewPipe/NewPipeExtractor#889) which will be supported by #10062.

> i have the newest dev Branch of both Projects

> I checked the git history of StreamInfoItem for the thumbnailUrl and the last change was in July 2022

Your statements are contradicting themselves, as there is no `thumbnailUrl` attribute for more than one month. Be sure that you checked out the commit corresponding to the one indicated in the Gradle app build file on your extractor clone, you should be then able to build the app.

**Comment by 0verEngineer** [comment]
@AudricV  Thanks a lot for your answer, however it does not help me with my problem and i am not able to re-open this discussion.

> as there is no thumbnailUrl attribute for more than one month

I have the latest dev of NewPipe and there are multiple references to thumbnailUrl, so i do not understand you statement?

> Be sure that you checked out the commit corresponding to the one indicated in the Gradle app build file on your extractor clone

I cannot find anything specific to any version / commit id in any of the Gradle files, not in the Extractor nor in NewPipe.


I want to build the latest NewPipe dev AND the NewPipe SponsorBlock Fork, so i need the Extractor version to match the 2 different NewPipes i build, i do not want the NewPipe version to match the Extractor version.

Thank you :)

**Comment by AudricV** [comment, marked as answer]
> I have the latest dev of NewPipe and there are multiple references to thumbnailUrl, so i do not understand you statement?

There no `thumbnailUrl` attribute in `StreamInfoItem` in the latest versions of the extractor, not in the version of the extractor currently used by the app.

> I cannot find anything specific to any version / commit id in any of the Gradle files, not in the Extractor nor in NewPipe.

See the following line: https://github.com/TeamNewPipe/NewPipe/blob/b798ff5c92eb5e8b740749b995eb7826fa8261dc/app/build.gradle#L200

The extractor commit used by the app is `95a3cc0a173bba28c179f9f9503b1010ec6bff21`, so you should use it in your local extractor this commit by running `git checkout 95a3cc0a173bba28c179f9f9503b1010ec6bff21` in the local extractor folder. You can also use any extractor commit which is compatible with the extractor API used by the app.

> I want to build the latest NewPipe dev AND the NewPipe SponsorBlock Fork, so i need the Extractor version to match the 2 different NewPipes i build, i do not want the NewPipe version to match the Extractor version.

The SponsorBlock fork uses [a different extractor version](https://github.com/polymorphicshade/NewPipe/blob/ed6a94b3fb41b605129e8ffc1efa989a390aa60c/app/build.gradle#L203), so I would use two clones of the extractor, one for the fork and one for the app with the correct versions being checked out.

As you can see in the fork (which is currently on top of the 0.25.2 version), we can also use release tags as extractor versions and not only commits.

**Reply by 0verEngineer** [reply]
Thanks a lot for the hint to the correct build.gradle file and the line, i can build it now :)

> There no thumbnailUrl attribute in StreamInfoItem in the latest versions of the extractor, not in the version of the extractor currently used by the app.

Yea i know, we just talked past each other, i was referring to the thumbnailUrl usage in the NewPipe App.
